src/deployment/7_claimChainInfo.js

```diff
@@ -1,105 +1,78 @@
+const networkName = hre.network.name;
+const { bridgeAPIEndpoint } = hre.config.zkEVMServices[networkName];
 
-const pathGenesisJson = path.join(__dirname, './genesis.json');
-const pathOutputJsonL2Applications = path.join(__dirname, './deploy_output_l2_applications.json');
-
-const genesisJSON = require(pathGenesisJson);
-const genesisEntries = genesisJSON.genesis;
-const l2Applications = require(pathOutputJsonL2Applications);
+const merkleProofString = '/merkle-proof';
+const getClaimsFromAcc = '/bridges/';
 
-let l2BridgeAddress;
-for(const genesisIdx in genesisEntries) {
-    const genesisEntry = genesisEntries[genesisIdx];
-    if (genesisEntry.contractName == "PolygonZkEVMBridge proxy") {
-        l2BridgeAddress = genesisEntry.address;
-        break;
-    }
-}
-if (!l2BridgeAddress) {
-    throw new Error('Could not find genesis bridge address in genesis.json');
+const baseURL = bridgeAPIEndpoint;
+if (!baseURL) {
+    throw new Error('Missing baseURL');
 }
 
-const merkleProofString = '/merkle-proof';
-const getClaimsFromAcc = '/bridges/';
+const axios = require('axios').create({
+    baseURL,
+});
 
-async function main() {
+const deployParameters = require('./deploy_application_parameters.json');
+const common = require('./common');
 
-    const currentProvider = ethers.provider;
-    let deployer;
-    if (process.env.PVTKEY) {
-        deployer = new ethers.Wallet(process.env.PVTKEY, currentProvider);
-        console.log('Using pvtKey deployer with address: ', deployer.address);
-    } else if (process.env.MNEMONIC) {
-        deployer = ethers.Wallet.fromMnemonic(process.env.MNEMONIC, 'm/44\'/60\'/0\'/0/0').connect(currentProvider);
-        console.log('Using MNEMONIC deployer with address: ', deployer.address);
-    } else {
-        [deployer] = (await ethers.getSigners());
-    }
+const l2Applications = require('./deploy_output_l2_applications.json');
 
-    const baseURL = bridgeAPIEndpoint;
-    if (!baseURL) {
-        throw new Error("Missing baseURL");
-    }
-    
-    const axios = require('axios').create({
-        baseURL,
-    });
+async function main() {
+    const l2BridgeAddress = common.genesisAddressForContractName('PolygonZkEVMBridge proxy');
 
-    const l2BridgeFactory = await ethers.getContractFactory('PolygonZkEVMBridge', deployer);
+    const currentProvider = await common.loadProvider(deployParameters, process.env);
+    const deployer = await common.loadDeployer(currentProvider, deployParameters);
+
+    const l2BridgeFactory = await ethers.getContractFactory('@RealityETH/zkevm-contracts/contracts/inheritedMainContracts/PolygonZkEVMBridge.sol:PolygonZkEVMBridge', deployer);
     const l2BridgeContract = l2BridgeFactory.attach(l2BridgeAddress);
 
-    const sleep = ms => new Promise(r => setTimeout(r, ms));
+    const sleep = (ms) => new Promise((r) => setTimeout(r, ms));
 
     function filterClaimable(_depositsArray, _verbose) {
```

Comment:
Maybe its easier to read if they are declared outside of main()?

Reply:
In javascript I like to keep function definitions close to where they're used with the most limited scope that'll do the job